### PR TITLE
Bloated output due to deprecation warning fixed

### LINKED ISSUE #244: Bloated testing output due to deprecated warnings
### Description
Reliance on deprecated functions, methods, etc. needs to be removed from out testing. 

### To reproduce

Just run the flow testing suite.

### Fix

Go through the testing output and find all references to deprecated functions and remove them.

### PR DESCRIPTION
**Pull request by kidrahahjo**
## Description
Placed warning filters on the deprecation warnings

## Motivation and Context

resolves #244 

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: contributors.yaml [hunk 1]
```diff
@@ -104,7 +104,13 @@ contributors:
     affiliation: "University of Michigan"
     orcid: "https://orcid.org/0000-0001-6443-7744"
   -
+<<<<<<< HEAD
```
Comment (bdice) [inline contributors.yaml]: Please resolve the merge conflicts here.

File: tests/test_project.py [hunk 2]
```diff
@@ -410,8 +416,8 @@ def test_context(job):
         project = A(self.mock_project().config)
         for job in project:
             job.doc.np = 3
-            next_op = project.next_operation(job)
-            assert 'mpirun -np 3 python' in next_op.cmd
+            next_op = project.next_operations(job)
```
Comment (bdice) [inline tests/test_project.py]: We haven't removed support for `next_operation` yet. We need to continue testing the deprecated feature until it is removed. You should add a warning filter and potentially add a second test that uses `next_operations`.

File: tests/test_project.py [hunk 3]
```diff
@@ -550,11 +557,12 @@ def test_next_operations(self):
                     assert op.name == 'op2'
             assert i == int(job in even_jobs)
 
+    @pytest.mark.filterwarnings("ignore:get_id")
     def test_get_job_status(self):
         project = self.mock_project()
         for job in project:
             status = project.get_job_status(job)
-            assert status['job_id'] == job.get_id()
+            assert status['job_id'] == job.id
```
Comment (bdice) [inline tests/test_project.py]: Leave this as `job.get_id()`.

File: tests/test_project.py [hunk 4]
```diff
@@ -663,34 +673,36 @@ def test_pending_operations_order(self):
...
-                output.seek(0)
-                output.read()
-                even_jobs = [job for job in project if job.sp.b % 2 == 0]
-                for job in project:
-                    if job in even_jobs:
-                        assert job.isfile('world.txt')
-                    else:
-                        assert not job.isfile('world.txt')
-
+            def sort_key(op):
```
Comment (bdice) [inline tests/test_project.py]: I don't think this block is supposed to be indented. Please revert.

File: tests/test_project.py [hunk 5]
```diff
@@ -1082,7 +1105,7 @@ def test_main_run(self):
     def test_main_next(self):
         assert len(self.project)
         jobids = set(self.call_subcmd('next op1').decode().split())
-        even_jobs = [job.get_id() for job in self.project if job.sp.b % 2 == 0]
+        even_jobs = [job.id for job in self.project if job.sp.b % 2 == 0]
```
Comment (bdice) [inline tests/test_project.py]: Revert this change and add a filter. Same with the other cases below.

**Comment by bdice** [comment]
@kidrahahjo Please leave the reviewers at the default (auto-assigned) next time, we try to distribute the load among the whole signac team and @csadorf and I have been getting a lot of review requests. There are 8 committers and about 8-10 contributors who help review pull requests. We typically assign one committer and one contributor.

File: tests/test_project.py (hunk 3 above)
Comment (kidrahahjo) [reply tests/test_project.py]: @bdice I did those changes in my commit 

> Fixed job.id errors created due to local testing

**Comment by kidrahahjo** [comment]
@bdice The get_id() function is used in flow/project module too.
For now, I've referred directly to the id variable instead of using a getter method.
Please let me know, if you're okay with that. Thanks

File: contributors.yaml [hunk 6]
```diff
@@ -104,13 +104,11 @@ contributors:
...
-    affiliation: "Indian Institute of Technology Roorkee"
-=======
     family-names: Sharma
     given-names: Vishav
     affiliation: "National Institute of Technology, Hamirpur"
->>>>>>> upstream/master
+  -
+    family-names: Ojha
+    given-names: Hardik
+    affiliation: "Indian Institue of Technology Roorkee"
```
Comment (bdice) [inline contributors.yaml]:
```suggestion
    affiliation: "Indian Institute of Technology Roorkee"
```

File: tests/test_project.py [hunk 7]
```diff
@@ -262,9 +262,9 @@ def label2(job):
         assert len(b._label_functions) == 2
         assert len(c._label_functions) == 1
 
+    @pytest.mark.filterwarnings("ignore:always")
     def test_conditions_with_inheritance(self):
         """Tests the inheritance of pre/post conditions.
-
```
Comment (bdice) [inline tests/test_project.py]: Add back the empty line you removed for compliance with PEP 257. Multiline docstrings should have a blank line. https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

File: tests/test_project.py [hunk 8]
```diff
@@ -1131,4 +1141,4 @@ def test_dag(self):
                        [0, 0, 0, 0, 0, 0, 0],
                        [0, 0, 0, 0, 0, 0, 0]]
 
-        assert adj == adj_correct
```
Comment (bdice) [inline tests/test_project.py]: All files should end with a newline character. It seems your editor may have trimmed this, or you removed it manually. Please revert this change.

File: tests/test_project.py [hunk 9]
```diff
@@ -262,9 +262,10 @@ def label2(job):
         assert len(b._label_functions) == 2
         assert len(c._label_functions) == 1
 
+    @pytest.mark.filterwarnings("ignore:always")
     def test_conditions_with_inheritance(self):
         """Tests the inheritance of pre/post conditions.
-
+
```
Comment (bdice) [inline tests/test_project.py]: @kidrahahjo Notice that you added whitespace here. I will remove it for you so that this PR can move forward.

**Comment by csadorf** [comment]
@bdice Is it not possible to suppress the warnings about `get_id`? The [test output](https://app.circleci.com/jobs/github/glotzerlab/signac-flow/7137/parallel-runs/0/steps/0-103) appears to be still pretty bloated to me.

**Comment by bdice** [comment]
@kidrahahjo I talked with @csadorf and he's going to take a closer look at the test output. There might be something else to fix. I'll let him reply here when he's had a chance to investigate.

**Comment by kidrahahjo** [comment]
> @bdice Is it not possible to suppress the warnings about `get_id`? The [test output](https://app.circleci.com/jobs/github/glotzerlab/signac-flow/7137/parallel-runs/0/steps/0-103) appears to be still pretty bloated to me.

@csadorf 
The bloated output is due to the model flow/project.py. 
When I referred directly to the variable in my machine then it worked quite well with only a few warnings left. But the tests failed when I pushed them

**Comment by csadorf** [comment]
> > @bdice Is it not possible to suppress the warnings about `get_id`? The [test output](https://app.circleci.com/jobs/github/glotzerlab/signac-flow/7137/parallel-runs/0/steps/0-103) appears to be still pretty bloated to me.
> 
> @csadorf
> The bloated output is due to the model flow/project.py.
> When I referred directly to the variable in my machine then it worked quite well with only a few warnings left. But the tests failed when I pushed them

I am going to have a deeper look at this, but I won't be able to do it anymore today.

**Comment by csadorf** [comment]
The test output is lean now: https://app.circleci.com/jobs/github/glotzerlab/signac-flow/7192

**Comment by csadorf** [comment]
@kidrahahjo Thank's a lot for fixing this!

**Comment by bdice** [comment]
@csadorf That was the right call -- thanks for the finishing touches.
